flink-runtime/src/test/java/org/apache/flink/runtime/executiongraph/failover/flip1/ExecutionFailureHandlerTest.java

```diff
@@ -167,23 +164,49 @@ public void testUnrecoverableErrorCheck() {
 			new Exception(new SuppressRestartsException(new Exception()))));
 	}
 
+	@Test
+	public void testGlobalFailureHandling() {
+		final FailureHandlingResult result = executionFailureHandler.getGlobalFailureHandlingResult(
+			new Exception("test failure"));
+
+		assertTrue(result.canRestart());
+		assertEquals(restartDelayMs, result.getRestartDelayMS());
+		assertEquals(
+			StreamSupport.stream(failoverTopology.getFailoverVertices().spliterator(), false)
+			.map(FailoverVertex::getExecutionVertexID)
+			.collect(Collectors.toSet()),
+			result.getVerticesToRestart());
+		try {
+			result.getError();
+			fail("Cannot get error when the restarting is accepted");
+		} catch (IllegalStateException ex) {
+			// expected
+		}
+	}
+
 	// ------------------------------------------------------------------------
 	//  utilities
 	// ------------------------------------------------------------------------
 
 	/**
-	 * A FailoverStrategy implementation for tests. It always suggest restarting the given task set on construction.
+	 * A FailoverStrategy implementation for tests. It always suggests restarting the given tasks to restart.
 	 */
 	private class TestFailoverStrategy implements FailoverStrategy {
 
-		private final Set<ExecutionVertexID> tasksToRestart;
+		private Set<ExecutionVertexID> tasksToRestart;
 
-		public TestFailoverStrategy(Set<ExecutionVertexID> tasksToRestart) {
-			this.tasksToRestart = checkNotNull(tasksToRestart);
+		public TestFailoverStrategy() {
+		}
+
+		public void setTasksToRestart(final Set<ExecutionVertexID> tasksToRestart) {
+			this.tasksToRestart = tasksToRestart;
```

Comment:
Consider making a defensive copy of the `Set` and returning an immutable Set in the getter. However, I am also fine with the current state since this is a test class.

Reply:
Ok. Will keep it as is since it is a private inner test class.